rust/arrow/src/array/equal/variable_size.rs

```diff
@@ -0,0 +1,90 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use crate::array::{ArrayData, OffsetSizeTrait};
+
+use super::utils::equal_len;
+
+fn offset_value_equal<T: OffsetSizeTrait>(
+    lhs_values: &[u8],
+    rhs_values: &[u8],
+    lhs_offsets: &[T],
+    rhs_offsets: &[T],
+    lhs_pos: usize,
+    rhs_pos: usize,
+    len: usize,
+) -> bool {
+    let lhs_start = lhs_offsets[lhs_pos].to_usize().unwrap();
+    let rhs_start = rhs_offsets[rhs_pos].to_usize().unwrap();
+    let lhs_len = lhs_offsets[lhs_pos + len] - lhs_offsets[lhs_pos];
+    let rhs_len = rhs_offsets[rhs_pos + len] - rhs_offsets[rhs_pos];
+
+    lhs_len == rhs_len
+        && equal_len(
+            lhs_values,
+            rhs_values,
+            lhs_start,
+            rhs_start,
+            lhs_len.to_usize().unwrap(),
+        )
+}
+
+pub fn variable_sized_equal<T: OffsetSizeTrait>(
```

Comment:
May you please add a doc, as it's a public function

Reply:
I have privatized this function, as it assumes that the user already used `base_equal`. This reminded me to document the `equal::equal` function, which is the public one of the module.